CONTRIBUTING.md

```diff
@@ -36,110 +39,144 @@ Signed-off-by: Jane Smith <jane.smith@example.com>
 
-
 ## <a name="contributing-meshery">Meshery Contribution Flow</a>
+
 Meshery is written in `Go` (Golang) and leverages Go Modules. UI is built on React and Next.js. To make building and packaging easier a `Makefile` is included in the main repository folder.
 
-Relevant coding style guidelines are the [Go Code Review Comments](https://code.google.com/p/go-wiki/wiki/CodeReviewComments) and the _Formatting and style_ section of Peter Bourgon's [Go: Best 
+Relevant coding style guidelines are the [Go Code Review Comments](https://code.google.com/p/go-wiki/wiki/CodeReviewComments) and the _Formatting and style_ section of Peter Bourgon's [Go: Best
 Practices for Production Environments](https://peter.bourgon.org/go-in-production/#formatting-and-style).
 
-__Please note__: All `make` commands should be run in a terminal from within the Meshery's main folder.
+**Please note**: All `make` commands should be run in a terminal from within the Meshery's main folder.
 
 ### Prerequisites for building Meshery in your development environment:
+
 1. `Go` version 1.11+ installed if you want to build and/or make changes to the existing code.
 1. `GOPATH` environment variable should be configured appropriately
 1. `npm` and `node` should be installed on your machine, preferably the latest versions.
 1. Fork this repository (`git clone https://github.com/layer5io/meshery.git`), clone your forked version of Meshery to your local, preferably outside `GOPATH`. If you happen to checkout Meshery inside your `GOPATH` and you have a version of `Go` prior to version 1.13, please set an environment variable `GO111MODULE=on` to enable GO Modules.
 
 #### Build and run Meshery server
+
+Before you can access the Meshery UI, you need to install the UI dependencies,
+
+```sh
+make setup-ui-libs
+```
+
+and then Build and export the UI
+
+```sh
+make build-ui
+```
+
 To build & run the Meshery server code, run the following command:
+
 ```sh
 make run-local
 ```
 
 Any time changes are made to the GO code, you will have to stop the server and run the above command again.
-Once the Meshery server is up and running, you should be able to access Meshery on your `localhost` on port `9081` at `http://localhost:9081`. One thing to note, you might NOT see the [Meshery UI](#contributing-ui) until the UI code is built as well.
-After running Meshery server, you will need to select your **Cloud Provider** by navigating to `localhost:9081`. Only then you will be able to use the Meshery UI on port `3000`.
+Once the Meshery server is up and running, you should be able to access Meshery on your `localhost` on port `9081` at `http://localhost:9081`.
+
+**NOTE**: You might NOT see the [Meshery UI](#contributing-ui) until the UI code is built as well.
+
+To access the Meshery UI Development Server port on `3000`, you will need to select your **Cloud Provider** by navigating to `localhost:9081` after running the Meshery server.
```

Comment:
Link the text `Development server` to the development server section.

Reply:
Done 🙂